src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -150,7 +150,7 @@ public function getValue($objectOrArray, $propertyPath)
             self::VALUE => $objectOrArray,
         ];
 
-        if (\is_object($objectOrArray) && false === strpbrk((string) $propertyPath, '.[')) {
+        if (\is_object($objectOrArray) && (false === strpbrk((string) $propertyPath, '.[') || property_exists($objectOrArray, $propertyPath))) {
```

Comment:
let's make this specific to stdClass?
this can be true for any object with dynamic properties but I don't think we need to support that, esp. if that costs too much, CPU-wise. Can you check this aspect.

Reply:
I've made it specific to `stdClass`. Here a quick benchmark, testing both accessing stdClass and other objects:

```php
<?php

use Symfony\Component\PropertyAccess\PropertyAccess;

require 'vendor/autoload.php';

class Foo
{
    public $iban = 'NL16TEST0436169118';
    public $bankSummary = '';
}

$stdclass = (object) ['iban' => 'NL16TEST0436169118', 'bankSummary' => ''];
$accessor = PropertyAccess::createPropertyAccessor();

dump("Testing with stdClass");
foreach (range(1, 5) as $i) {
    $start = microtime(true);

    for ($j = 0; $j < 100000; $j++) {
        $accessor->setValue($stdclass, 'iban', 'value');
        $accessor->getValue($stdclass, 'iban');
    }

    $end = microtime(true);

    dump('Elapsed time: ' . ($end - $start) . ' seconds');
}

dump("\nTesting with Foo");
$foo = new Foo();
foreach (range(1, 5) as $i) {
    $start = microtime(true);

    for ($j = 0; $j < 100000; $j++) {
        $accessor->setValue($foo, 'iban', 'value');
        $accessor->getValue($foo, 'iban');
    }

    $end = microtime(true);

    dump('Elapsed time: ' . ($end - $start) . ' seconds');
}
```

Above is **with** the patch, below it 5.4 as is:

![image](https://github.com/user-attachments/assets/c8f4ac54-220c-47df-9ddf-80a5aaa90eb6)

The difference is low for 100 000 iterations. After running the script several times, the time differences are explained more by external tasks that take up some CPU during execution.